src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1979,23 +2024,213 @@ def generate(
             if kwargs.get("num_frames") is not None:
                 generation_config.num_frames = kwargs.pop("num_frames")
 
-        outputs = super().generate(
-            inputs,
-            generation_config,
-            logits_processor,
-            stopping_criteria,
-            prefix_allowed_tokens_fn,
-            synced_gpus,
-            **kwargs,
-        )
+        if generation_config.return_timestamps is True:
+            last_forced_decoder_ids = (
```

Comment:
This was a bug previously, we need to be sure to remove `forced_decoder_ids` in case the last forced decoder id is set to "Don't use timestamp token id"